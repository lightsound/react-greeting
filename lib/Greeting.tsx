import React from "react";

export interface Props {
  text: string;
}

const Greeting: React.FC<Props> = ({ text }: Props) => {
  if (!text) return null;
  return <h1>{`Hello, ${text}!`}</h1>;
};

export default Greeting;
