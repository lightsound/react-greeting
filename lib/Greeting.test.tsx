import React from "react";
import { render } from "react-testing-library";
import Greeting from "./Greeting";

test("renders correctly", () => {
  const { asFragment } = render(<Greeting text="react" />);
  expect(asFragment()).toMatchSnapshot();
});
