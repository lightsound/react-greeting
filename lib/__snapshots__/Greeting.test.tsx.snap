// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <h1>
    Hello, react!
  </h1>
</DocumentFragment>
`;
